README.md

```diff
@@ -56,44 +56,83 @@ numbers as plain text, but demonstrates a potential practical use case for the c
   - ` Mat& roi(int dim1Start = -1, int dim1End = -1, int dim2Start = -1, int dim2End = -1) `
 ### Modifiers
 - **operator+**: elementwise addition
-  - ` Mat operator+(const Mat<Type>&) `
+  - ` Mat<Type> operator+(const Mat<Type>&) `
+  - ` Mat<Type> operator+(const Type) `
+  - ` Mat<Type> operator+(const Type, const Mat<Type>&) `
+- **operator+=**: elementwise addition and assignment
+  - ` void operator+=(const Mat<Type>&) `
 - **operator-**: elementwise subtraction, or negation when used as unary operator
-  - ` Mat operator-(const Mat<Type>&) `
-  - ` Mat operator-() `
+  - ` Mat<Type> operator-(const Mat<Type>&) `
+  - ` Mat<Type> operator-(const Type) `
+  - ` Mat<Type> operator-(const Type, const Mat<Type>&) `
+  - ` Mat<Type> operator-() `
+- **operator-=**: elementwise subtraction and assignment
+  - ` void operator-(const Mat<Type>&) `
+- **operator\***: elementwise multiplication
+  - ` Mat<Type> operator*(const Mat<Type>&) `
+  - ` Mat<Type> operator*(const Type) `
+  - ` Mat<Type> operator*(const Type, const Mat<Type>&) `
+- **operator\*=**: elementwise multiplication and assignment
+  - ` void operator*=(const Mat<Type>&) `
+- **operator/**: elementwise division
+  - ` Mat<Type> operator/(const Mat<Type>&) `
+  - ` Mat<Type> operator/(const Type) `
+  - ` Mat<Type> operator/(const Type, const Mat<Type>&) `
+- **operator/=**: elementwise division and assignment
+  - ` void operator/=(const Mat<Type>&) `
 - **broadcast**: binary operation which applies a given elementwise function and returns resulting matrix. Imitates NumPy array broadcasting.
-  - ` Mat broadcast(const Mat<Type>&, Type (*f)(Type,Type)) `
-- **operator^**: matrix multiplication
-  - ` Mat operator^(const Mat<Type>&) `
-- **T**: returns transposed matrix if no arguments are given, otherwise stores transpose into given destination matrix
-  - ` Mat T() `
-  - ` void T(Mat&) `
+  - ` Mat<Type3> broadcast(const Mat<Type2>&, Type3 (*f)(Type,Type2)) `
+  - ` Mat<Type3> broadcast(Type2, Type3 (*f)(Type,Type2)) `
+- **static broadcast**: static version in case matrix must be right operand
+  - ` Mat<Type3> broadcast(Type, const Mat<Type2>&, Type3 (*f)(Type,Type2)) `
+  - ` Mat<Type3> broadcast(Type2, Type3 (*f)(Type,Type2)) `
+- **operator^**: matrix multiplication (Not XOR!)
```

Comment:
comment very early in usage section that says mostly normal except xor chosen for matrix multiply

Reply:
mention that we try to follow numpy except for transpose, roi, and matrix multiply operators